### ISSUE TITLE
PS3 controller (DualShock 3) missing axes

### ISSUE THREAD

**Issue by rodjun** [pygame/pygame#338]
I'm using a ps3 controller with PSNOW drivers and pygame is not detected the axis used by the right trigger, it works fine with pygame_sdl2

Joystick example with pygame (6 axes detected, right trigger is not in the list):
http://i.imgur.com/Um9asiM.png

Chaing the first line to import pygame_sdl2 as pygame (8 axes, right trigger is there):
http://i.imgur.com/P7LT7Tu.png

**Comment by adicarlo** [comment]
Do you have a small / standalone python snippet to show the issue?

**Comment by rodjun** [comment]
The one i used in my pictures is at the end of [this](https://www.pygame.org/docs/ref/joystick.html) page, you can see the difference by just changing "import pygame" to "import pygame_sdl2 as pygame". 

I can also build a simple one that imports both and shows the number of axes detected or any other info that you find useful.

**Comment by adicarlo** [comment]
Yeah, I'm not a core developer, just looking for bugs to knock off.

The small example program might be helpful to the developers.  Also helpful is to paste text results, as in your first comments, in text areas rather than as images.  Text can be diff'ed and handled in all sorts of ways.

For my part, just puzzling over how to attack this without having the hardware.

**Comment by adicarlo** [comment]
Might also be a good idea to retitle this bug:

  PS3 controller missing 2 axis

I don't seem to have the perms.

**Comment by rodjun** [comment]
Just did it, thanks for the heads up.

**Comment by adicarlo** [comment]
@rodjun +1 for the corrected spelling!  I'll dig around when I get a free mental cycle, using some of the joysticks I do have (ps2, and a couple rumblepad variations).

**Comment by illume** [comment]
Seems like this will be fixed with an SDL2 pygame.

**Comment by powerthinking** [comment]
It's not just the PS3 controller that is having the issue.  I am only getting only 2 axes (get_numaxes) for a racing wheel controller that has 8, which I am properly seeing on jstest-gtk tool.  

Any idea when this will be resolved?

![pygame-axes-bug](https://user-images.githubusercontent.com/16319465/51366131-dfee9800-1ab1-11e9-83a6-a15bf6acbd30.jpg)

![joystick-axes](https://user-images.githubusercontent.com/16319465/51366079-9b62fc80-1ab1-11e9-9c1c-aeee73df74eb.jpg)

![joystick-axes-details](https://user-images.githubusercontent.com/16319465/51366083-9dc55680-1ab1-11e9-9340-ba232866e63f.jpg)

jstest-gtk: https://github.com/Grumbel/jstest-gtk

**Comment by MyreMylar** [comment]
Fairly sure that most of this bug is wrong as there are only 6 axis on a standard controller:

1. **Right Stick**: Left <->Right.
2. **Right Stick:** Up <-> Down.
3. **Left Stick:** Left <->Right.
4. **Left Stick:** Up <-> Down.
5. **Right Trigger:** In <-> Out.
6. **Left Trigger:** In <-> Out.

I've just tested the joystick example in 2.0.0.dev8 and 1.9.6 with both a XBox 360 controller and a PS4 controller on Windows and they both report the correct number of axii, which is six. 

There is no special functionality in pygame around the number of axis in controllers it is all just handled by SDL_JoystickGetAxis & SDL_JoystickNumAxes. 

There is a new game controller API in SDL but I think that will be eventually exist in a separate pygame module so as not to break compatability.

TLDR; I think this should probably be closed.
